src/analysis/calculus/uniform_limits_deriv.lean

```diff
@@ -0,0 +1,394 @@
+  derivatives `f' : ℕ → (E → (E →L[𝕜] G))` and the `f` converge pointwise to `g` and the `f'`
+  converge _uniformly_ on some closed ball, then the derivative of `g'` is the pointwise limit
+  of the `f'` on the closed ball
+
+## Implementation notes
+
+Our proof utilizes three major components:
+  * `convex.norm_image_sub_le_of_norm_has_fderiv_within_le`: The mean value inequality for
+    vector-valued functions over `ℝ` and `ℂ`
+  * `norm_add_le`: The triangle inequality
+  * `uniform_cauchy_seq_on.tendsto_uniformly_on_of_tendsto` which allows us to upgrade pointwise
+    convergence to uniform convergence by showing that the Cauchy sequences converge uniformly to 0
+
+## Tags
+
+uniform convergence, limits of derivatives
+-/
+
+open finset filter metric
+open_locale uniformity filter
+
+section convenience_lemmas
+
+/-- Convenience lemma -/
+private lemma mul_cancel_inv_left₀ {a b : ℝ} (ha : a ≠ 0) : a⁻¹ * (a * b) = b :=
+ begin
+  conv { congr, congr, skip, rw ←inv_inv a, },
+  have : a⁻¹ ≠ 0, simp [ha],
+  rw mul_inv_cancel_left₀ this,
+end
+
+/-- We'll need to apply the triangle ineqaulity twice, which is easier to do in lemma
+format -/
+private lemma norm_add_three_le {E : Type*} [semi_normed_group E] {a b c : E} :
+  ∥a + b + c∥ ≤ ∥a∥ + ∥b∥ + ∥c∥ :=
+begin
+  refine le_trans (norm_add_le _ _) _,
+  exact add_le_add (norm_add_le _ _) rfl.le,
+end
+
+/-- Convenience lemma for three way comparisons -/
+private lemma lt_of_lt_of_lt_of_le {a b c a' b' c': ℝ} :
+  a < a' → b < b' → c ≤ c' → a + b + c < a' + b' + c' :=
+begin
+  intros ha hb hc,
+  linarith [ha, hb, hc],
+end
+
+end convenience_lemmas
+
+section limits_of_derivatives
+
+variables {E : Type*} [normed_group E] {𝕜 : Type*} {G : Type*}
+  [is_R_or_C 𝕜] [normed_group G] [normed_space 𝕜 G]
+  {f : ℕ → E → G} {g : E → G} {x y z : E} {r R C : ℝ}
+
+/-- For fixed y and z, if `f_n → g` pointwise, then the difference quotients
+`∥z - y∥⁻¹ • (f_n z - f_n y)` converge to the difference quotients
+`∥z - y∥⁻¹ • (g z - g y)` -/
+lemma difference_quotients_converge
+  (hfg : ∀ (y : E), y ∈ closed_ball x r → tendsto (λ n, f n y) at_top (nhds (g y))) :
+  ∀ y : E, y ∈ closed_ball x r → ∀ z : E, z ∈ closed_ball x r →
+    tendsto (λ n : ℕ, (∥z - y∥⁻¹ : 𝕜) • (f n z - f n y))
+      at_top (nhds ((∥z - y∥⁻¹ : 𝕜) • (g z - g y))) :=
+(λ y hy z hz, ((hfg z hz).sub (hfg y hy)).const_smul _)
+
+variables [normed_space ℝ E] [normed_space 𝕜 E] {f' : ℕ → (E → (E →L[𝕜] G))} {g' : E → (E →L[𝕜] G)}
+
+/-- A convenience theorem for utilizing the mean value theorem for differences of
+differentiable functions -/
+lemma mean_value_theorem_for_differences {f : E → G} {f' : E → (E →L[𝕜] G)}
+  (hrR : r < R) (hf : ∀ (y : E), y ∈ ball x R → has_fderiv_at f (f' y) y)
+  (hg : ∀ (y : E), y ∈ ball x R → has_fderiv_at g (g' y) y)
+  (hbound : ∀ (y : E), y ∈ closed_ball x r → ∥f' y - g' y∥ ≤ C)
+  (hy : y ∈ closed_ball x r) (hz : z ∈ closed_ball x r) :
+  ∥y - z∥⁻¹ * ∥(f y - g y) - (f z - g z)∥ ≤ C :=
```

Comment:
Why do you assume `hf` and `hg` on a larger set than `closed_ball x r`? The proof goes through just fine (with trivial modifications) if I replace `ball x R` by `closed_ball x r`.
Moreover, can you generalize `closed_ball x r` to a more general set? What do you really need about this set? Does it need to be closed? A neighborhood of `x`? convex? Generalize this lemma and the lemmas below.